wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubURLParser.java

```diff
@@ -88,12 +97,14 @@ public GithubUrl parse(String url) {
       }
     }
 
-    return new GithubUrl()
-        .withUsername(repoUser)
-        .withRepository(repoName)
-        .withBranch(branchName)
-        .withSubfolder(matcher.group("subFolder"))
-        .withDevfileFilename("devfile.yaml")
-        .withFactoryFilename(".factory.json");
+    GithubUrl githubUrl =
+        new GithubUrl()
+            .withUsername(repoUser)
+            .withRepository(repoName)
+            .withBranch(branchName)
+            .withFactoryFilename(".factory.json")
+            .withSubfolder(matcher.group("subFolder"));
+    githubUrl.withDevfileFilenames(devfileFilenamesProvider.getConfiguredDevfileFilenames());
```

Comment:
so chain it with others and just return immediately without `    GithubUrl githubUrl =` ? :)